kvpress/presses/think_press.py

```diff
@@ -104,3 +103,16 @@ def forward_hook(self, module: nn.Module, input: list[torch.Tensor], kwargs: dic
             cache.key_cache[module.layer_idx] = keys
 
         return output
+
+    @property
+    def compression_ratio(self):
+        compression_ratio = self.key_channel_compression_ratio / 2
+        if self.inner_press is not None and hasattr(self.inner_press, "compression_ratio"):
```

Comment:
remove the inner_press and simply return self.key_channel_compression_ratio / 2. inner_press is a more generic press composition operation that can be implemented in another class (will open an issue on it)

Reply:
Covered by https://github.com/NVIDIA/kvpress/pull/29
Let's maybe work on this PR first, merge it and then update and merge #29 directly afterwards.